### PR TITLE
Add server GAMEPLAY_STARTING state.

### PR DESCRIPTION
**Pull request by karliss**
Fixes [2962](https://bugs.vcmi.eu/view.php?id=2962) .

### CODE REVIEW

File: server/CVCMIServer.h [hunk 1]
```diff
@@ -9,6 +9,8 @@
  */
 #pragma once
 
+#include "Global.h"
```
Comment (ArseniyShestakov) [inline server/CVCMIServer.h]: This should not be required I suppose.

File: server/NetPacksLobbyServer.cpp [hunk 2]
```diff
@@ -115,6 +115,9 @@ bool LobbyChatMessage::checkClientPermissions(CVCMIServer * srv) const
 
 bool LobbySetMap::applyOnServer(CVCMIServer * srv)
 {
+	if (srv->state != EServerState::LOBBY) {
+		return false;
+	}
```
Comment (ArseniyShestakov) [inline server/NetPacksLobbyServer.cpp]: Should be formatted according to our guidelines.

File: server/CVCMIServer.h (hunk 1 above)
Comment (karliss) [reply server/CVCMIServer.h]: Removed it, lets fix it globally.

File: server/NetPacksLobbyServer.cpp (hunk 2 above)
Comment (karliss) [reply server/NetPacksLobbyServer.cpp]: Done

**Comment by ArseniyShestakov** [comment]
Ok I cherry-picked it again so not to waste your time:
https://github.com/vcmi/vcmi/commit/965b33d16a5e98d334e401e558fde7be9d56acf9

**Pull request closed without merging**